aws/src/main/java/org/apache/iceberg/aws/s3/S3RequestUtil.java

```diff
@@ -149,4 +151,10 @@ static void configurePermission(
       Function<ObjectCannedACL, S3Request.Builder> aclSetter) {
     aclSetter.apply(s3FileIOProperties.acl());
   }
+
+  // TODO Remove me once all of the S3-compatible storage support strong integrity checks
```

Comment:
I understand the intent but this doesn't feel like a great way of going about this just because there's many out there and "all" just seems like a moving goalpost.